File: counting/#242_valid_anagram/isAnagram.go
package main

import "fmt"

func main() {
	isAnagram("anagram", "nagaram") // true
	isAnagram("rat", "cat")         // false
	isAnagram("a", "abc")           // false
}

func isAnagram(s string, t string) bool {
	chars := make([]int, 26)

	for _, v := range s {
		i := int(v - 'a')
		chars[i]++
	}

	for _, v := range t {
		i := int(v - 'a')
		chars[i]--
	}

	for _, v := range chars {
		if v != 0 {
			fmt.Println(false)
			return false
		}
	}

	fmt.Println(true)
	return true
}

File: counting/#242_valid_anagram/isAnagram.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isAnagram = function (s, t) {
  if(s.length !== t.length) {
    console.log(false);
    return false
  }

  const counter = {}
  for(let e of s) {
    counter[e] = (counter[e] || 0) + 1
  }

  for(let e of t) {
    if(!counter[e]) {
      console.log(false);
      return false
    }
    counter[e]--
  }

  console.log(true);
  return true
};


isAnagram("anagram", "nagaram")  // true
isAnagram("rat", "cat")          // false
isAnagram("a", "abc")            // false

